compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -525,6 +533,16 @@ macro_rules! define_queries {
                 })
             }
 
+            #[inline]
+            fn loadable_from_disk(_qcx: QueryCtxt<'tcx>, _key: &Self::Key, _index: SerializedDepNodeIndex) -> bool {
```

Comment:
In code: if `on_disk_cache` is None, or if `_index` is not in the cached result.

The former means no incremental, so the distinction shouldn't happen.

For the latter, I still don't understand how that's possible. A corrupt/empty on-disk cache would ICE in plenty other places, so that's not it.

Something missing in cache promotion?